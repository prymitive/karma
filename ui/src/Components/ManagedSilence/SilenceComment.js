import React from "react";
import PropTypes from "prop-types";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faExternalLinkAlt } from "@fortawesome/free-solid-svg-icons/faExternalLinkAlt";
import { faBellSlash } from "@fortawesome/free-solid-svg-icons/faBellSlash";

import { APISilence } from "Models/API";
import { AlertStore } from "Stores/AlertStore";
import { FilteringCounterBadge } from "Components/Labels/FilteringCounterBadge";
import { ToggleIcon } from "Components/ToggleIcon";
import { SilenceProgress } from "./SilenceProgress";

const SilenceComment = ({
  cluster,
  silence,
  alertCount,
  alertCountAlwaysVisible,
  collapsed,
  collapseToggle,
  afterUpdate,
  alertStore,
}) => {
  const comment = silence.ticketURL ? (
    <a
      href={silence.ticketURL}
      target="_blank"
      rel="noopener noreferrer"
      className={
        collapsed
          ? "components-managed-silence-comment mw-100 text-truncate d-block"
          : "components-managed-silence-comment"
      }
    >
      <FontAwesomeIcon className="mr-2" icon={faExternalLinkAlt} />
      {silence.comment}
    </a>
  ) : (
    silence.comment
  );

  return (
    <React.Fragment>
      <div className="d-flex flex-row">
        <div className="flex-shrink-0 flex-grow-0">
          <FontAwesomeIcon
            icon={faBellSlash}
            className="components-managed-silence-icon text-muted"
          />
        </div>
        <div className="mx-2 flex-shrink-1 flex-grow-1 mw-1p">
          <div
            className={`font-italic components-managed-silence-comment ${
              collapsed ? "text-truncate overflow-hidden" : ""
            }`}
          >
            {comment}
          </div>
          <div className="components-managed-silence-cite mt-1">
            <span className="text-muted mr-2 font-italic">
              &mdash; {silence.createdBy}
            </span>
            {collapsed &&
            Object.keys(alertStore.data.upstreams.clusters).length > 1 ? (
              <span className="badge badge-secondary mx-1 align-text-bottom p-1">
                {cluster}
              </span>
            ) : null}
            {collapsed ? <SilenceProgress silence={silence} /> : null}
          </div>
        </div>
        <div className="flex-shrink-0 flex-grow-0">
          <div className="d-flex flex-column flex-sm-row justify-content-between align-items-center">
            <FilteringCounterBadge
              alertStore={alertStore}
              name="@silence_id"
              value={silence.id}
              counter={alertCount}
              themed={false}
              alwaysVisible={alertCountAlwaysVisible}
              defaultColor="primary"
            />
            <ToggleIcon
              isOpen={!collapsed}
              className={`components-managed-silence-icon ${
                alertCount && alertCountAlwaysVisible && "my-sm-auto mt-2 mb-0"
              } ml-sm-2 ml-auto mr-sm-0 mr-1 text-muted cursor-pointer`}
              onClick={collapseToggle}
            />
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};
SilenceComment.propTypes = {
  cluster: PropTypes.string.isRequired,
  silence: APISilence.isRequired,
  alertCount: PropTypes.number.isRequired,
  collapsed: PropTypes.bool.isRequired,
  collapseToggle: PropTypes.func.isRequired,
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
};

export { SilenceComment };
