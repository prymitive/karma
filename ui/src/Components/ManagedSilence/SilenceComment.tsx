import { FC } from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faExternalLinkAlt } from "@fortawesome/free-solid-svg-icons/faExternalLinkAlt";
import { faBellSlash } from "@fortawesome/free-solid-svg-icons/faBellSlash";

import { APISilenceT } from "Models/APITypes";
import { AlertStore } from "Stores/AlertStore";
import FilteringCounterBadge from "Components/Labels/FilteringCounterBadge";
import { ToggleIcon } from "Components/ToggleIcon";
import { SilenceProgress } from "./SilenceProgress";

const SilenceComment: FC<{
  cluster: string;
  silence: APISilenceT;
  alertCount: number;
  collapsed: boolean;
  collapseToggle: () => void;
  alertStore: AlertStore;
  alertCountAlwaysVisible?: boolean;
}> = ({
  cluster,
  silence,
  alertCount,
  alertCountAlwaysVisible,
  collapsed,
  collapseToggle,
  alertStore,
}) => {
  const comment = silence.comment.split(" ").map((w, i) =>
    silence.ticketURL && w === silence.ticketID ? (
      <a
        key={i}
        href={silence.ticketURL}
        target="_blank"
        rel="noopener noreferrer"
      >
        <FontAwesomeIcon className="me-2" icon={faExternalLinkAlt} />
        {silence.ticketID}
      </a>
    ) : (
      " " + w
    )
  );

  return (
    <>
      <div className="d-flex flex-row">
        <div className="flex-shrink-0 flex-grow-0">
          <FontAwesomeIcon
            icon={faBellSlash}
            className="components-managed-silence-icon text-muted"
          />
        </div>
        <div className="mx-2 flex-shrink-1 flex-grow-1 mw-1p">
          <div
            className={`font-italic components-managed-silence-comment ${
              collapsed ? "text-truncate overflow-hidden" : ""
            }`}
          >
            {comment.map((w) => w)}
          </div>
          <div className="components-managed-silence-cite mt-1">
            <span className="text-muted me-2 font-italic">
              &mdash; {silence.createdBy}
            </span>
            {collapsed &&
            Object.keys(alertStore.data.upstreams.clusters).length > 1 ? (
              <span className="badge bg-secondary mx-1 align-text-bottom p-1">
                {cluster}
              </span>
            ) : null}
            {collapsed ? <SilenceProgress silence={silence} /> : null}
          </div>
        </div>
        <div className="flex-shrink-0 flex-grow-0">
          <div className="d-flex flex-column flex-sm-row justify-content-between align-items-center">
            <FilteringCounterBadge
              alertStore={alertStore}
              name="@silence_id"
              value={silence.id}
              counter={alertCount}
              themed={false}
              alwaysVisible={alertCountAlwaysVisible}
              defaultColor="primary"
              isAppend={false}
            />
            <span className="badge components-label with-click">
              <ToggleIcon
                isOpen={!collapsed}
                className="components-managed-silence-icon m-auto text-muted cursor-pointer"
                onClick={collapseToggle}
              />
            </span>
          </div>
        </div>
      </div>
    </>
  );
};

export { SilenceComment };
