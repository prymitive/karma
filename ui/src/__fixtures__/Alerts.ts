import {
  APIAlertmanagerUpstreamT,
  APIAlertGroupT,
  APIAlertT,
  APISilenceT,
  APIAnnotationT,
  AlertStateT,
  LabelsT,
} from "Models/APITypes";

const MockAnnotation = (
  name: string,
  value: string,
  visible: boolean,
  isLink: boolean,
  isAction: boolean
): APIAnnotationT => ({
  name: name,
  value: value,
  visible: visible,
  isLink: isLink,
  isAction: isAction,
});

const MockAlert = (
  annotations: APIAnnotationT[],
  labels: LabelsT,
  state: AlertStateT
): APIAlertT => ({
  id: Math.random().toString(36),
  annotations: annotations,
  labels: labels,
  startsAt: "2018-08-14T17:36:40.017867056Z",
  state: state,
  alertmanager: [
    {
      fingerprint: "1234567",
      name: "default",
      cluster: "default",
      state: state,
      startsAt: "2018-08-14T17:36:40.017867056Z",
      source: "localhost/graph",
      silencedBy: [],
      inhibitedBy: [],
    },
  ],
  receiver: "by-name",
});

const MockAlertGroup = (
  rootLabels: LabelsT,
  alerts: APIAlertT[],
  sharedAnnotations: APIAnnotationT[],
  sharedLabels: LabelsT,
  sharedSilences: { [cluster: string]: string[] }
): APIAlertGroupT => ({
  receiver: "by-name",
  labels: rootLabels,
  alerts: alerts,
  id: "099c5ca6d1c92f615b13056b935d0c8dee70f18c",
  alertmanagerCount: {
    default: 1,
  },
  stateCount: {
    active: 1,
    suppressed: 0,
    unprocessed: 0,
  },
  shared: {
    annotations: sharedAnnotations,
    labels: sharedLabels,
    silences: sharedSilences,
  },
});

const MockSilence = (): APISilenceT => ({
  comment: "Mocked Silence",
  createdAt: "0001-01-01T00:00:00Z",
  createdBy: "me@example.com",
  startsAt: "2000-01-01T00:00:00Z",
  endsAt: "2000-01-01T01:00:00Z",
  id: "04d37636-2350-4878-b382-e0b50353230f",
  ticketID: "",
  ticketURL: "",
  matchers: [
    { name: "regex", value: "equal", isRegex: true, isEqual: true },
    { name: "regex", value: "notEqual", isRegex: true, isEqual: false },
    { name: "notRegex", value: "equal", isRegex: false, isEqual: true },
    { name: "notRegex", value: "notEqual", isRegex: false, isEqual: false },
  ],
});

const MockAlertmanager = (): APIAlertmanagerUpstreamT => ({
  name: "default",
  cluster: "default",
  uri: "http://localhost",
  publicURI: "http://am.example.com",
  readonly: false,
  headers: {
    Authorization: "Basic foo bar",
  },
  corsCredentials: "include",
  error: "",
  version: "0.17.0",
  clusterMembers: ["default"],
});

export {
  MockAnnotation,
  MockAlert,
  MockAlertGroup,
  MockSilence,
  MockAlertmanager,
};
