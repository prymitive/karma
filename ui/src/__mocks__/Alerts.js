const MockAnnotation = (name, value, visible, isLink) => ({
  name: name,
  value: value,
  visible: visible,
  isLink: isLink,
});

const MockAlert = (annotations, labels, state) => ({
  id: Math.random().toString(36),
  annotations: annotations,
  labels: labels,
  startsAt: "2018-08-14T17:36:40.017867056Z",
  state: state,
  alertmanager: [
    {
      name: "default",
      cluster: "default",
      state: "active",
      startsAt: "2018-08-14T17:36:40.017867056Z",
      source: "localhost/prometheus",
      silencedBy: [],
      inhibitedBy: [],
    },
  ],
  receiver: "by-name",
});

const MockAlertGroup = (
  rootLabels,
  alerts,
  sharedAnnotations,
  sharedLabels,
  sharedSilences
) => ({
  receiver: "by-name",
  labels: rootLabels,
  alerts: alerts,
  id: "099c5ca6d1c92f615b13056b935d0c8dee70f18c",
  alertmanagerCount: {
    default: 1,
  },
  stateCount: {
    active: 1,
    suppressed: 0,
    unprocessed: 0,
  },
  shared: {
    annotations: sharedAnnotations,
    labels: sharedLabels,
    silences: sharedSilences,
  },
});

const MockSilence = () => ({
  comment: "Mocked Silence",
  createdAt: "0001-01-01T00:00:00Z",
  createdBy: "me@example.com",
  startsAt: "2000-01-01T00:00:00Z",
  endsAt: "2000-01-01T01:00:00Z",
  id: "04d37636-2350-4878-b382-e0b50353230f",
  ticketID: "",
  ticketURL: "",
  matchers: [
    { name: "foo", value: "bar", isRegex: false },
    { name: "baz", value: "regex", isRegex: true },
  ],
});

const MockAlertmanager = () => ({
  name: "default",
  cluster: "default",
  uri: "http://localhost",
  publicURI: "http://am.example.com",
  readonly: false,
  headers: {
    Authorization: "Basic foo bar",
  },
  corsCredentials: "include",
  error: "",
  version: "0.17.0",
  clusterMembers: ["default"],
});

export {
  MockAnnotation,
  MockAlert,
  MockAlertGroup,
  MockSilence,
  MockAlertmanager,
};
