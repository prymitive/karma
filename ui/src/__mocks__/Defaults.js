const DefaultsBase64 =
  "eyJSZWZyZXNoIjo0NTAwMDAwMDAwMCwiSGlkZUZpbHRlcnNXaGVuSWRsZSI6ZmFsc2UsIkNvbG9yVGl0bGViYXIiOmZhbHNlLCJNaW5pbWFsR3JvdXBXaWR0aCI6NTU1LCJBbGVydHNQZXJHcm91cCI6MTUsIkNvbGxhcHNlR3JvdXBzIjoiZXhwYW5kZWQifQo=";
const DefaultsObject = {
  Refresh: 45000000000,
  HideFiltersWhenIdle: false,
  ColorTitlebar: false,
  MinimalGroupWidth: 555,
  AlertsPerGroup: 15,
  CollapseGroups: "expanded"
};

export { DefaultsBase64, DefaultsObject };
