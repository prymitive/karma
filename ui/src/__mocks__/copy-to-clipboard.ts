// mock copy-to-clipboard since it throws errors in tests
// and we don't really need to copy anything, only ensure we're calling it

const copy = jest.fn();

export default copy;
