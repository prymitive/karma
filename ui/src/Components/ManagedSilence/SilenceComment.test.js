import React from "react";

import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { MockSilence } from "__mocks__/Alerts";
import { AlertStore } from "Stores/AlertStore";
import { SilenceComment } from "./SilenceComment";

let silence;
let alertStore;

beforeEach(() => {
  silence = MockSilence();
  alertStore = new AlertStore([]);
});

afterEach(() => {
  jest.restoreAllMocks();
  fetch.resetMocks();
});

const CollapseMock = jest.fn();

const MountedSilenceComment = (collapsed) => {
  return mount(
    <SilenceComment
      alertStore={alertStore}
      alertCount={123}
      cluster="default"
      silence={silence}
      collapsed={collapsed}
      collapseToggle={CollapseMock}
    />
  );
};

const MockMultipleClusters = () => {
  alertStore.data.upstreams = {
    clusters: { default: ["default", "fallback"], second: ["second"] },
    instances: [
      {
        name: "default",
        uri: "http://am1.example.com",
        publicURI: "http://am1.example.com",
        readonly: false,
        headers: {},
        corsCredentials: "include",
        error: "",
        version: "0.17.0",
        cluster: "default",
        clusterMembers: ["default", "fallback"],
      },
      {
        name: "fallback",
        uri: "http://am2.example.com",
        publicURI: "http://am2.example.com",
        readonly: false,
        headers: {},
        corsCredentials: "include",
        error: "",
        version: "0.17.0",
        cluster: "default",
        clusterMembers: ["default", "fallback"],
      },
      {
        name: "second",
        uri: "http://am3.example.com",
        publicURI: "http://am3.example.com",
        readonly: false,
        headers: {},
        corsCredentials: "include",
        error: "",
        version: "0.17.0",
        cluster: "second",
        clusterMembers: ["second"],
      },
    ],
  };
};

describe("<SilenceComment />", () => {
  it("Matches snapshot when collapsed", () => {
    const tree = MountedSilenceComment(true);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Matches snapshot when expanded", () => {
    const tree = MountedSilenceComment(false);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Matches snapshot when collapsed and multiple clusters are present", () => {
    MockMultipleClusters();
    const tree = MountedSilenceComment(true);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Matches snapshot when collapsed and multiple clusters are present", () => {
    MockMultipleClusters();
    const tree = MountedSilenceComment(false);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Renders a JIRA link if present", () => {
    silence.ticketURL = "http://localhost/1234";
    silence.ticketID = "1234";
    const tree = MountedSilenceComment(true);
    expect(tree.find("a[href='http://localhost/1234']")).toHaveLength(1);
  });

  it("Renders a JIRA link if present and comment is expanded", () => {
    silence.ticketURL = "http://localhost/1234";
    silence.ticketID = "1234";
    const tree = MountedSilenceComment(false);
    expect(tree.find("a[href='http://localhost/1234']")).toHaveLength(1);
  });

  it("collapseToggle is called when collapse icon is clicked", () => {
    const tree = MountedSilenceComment(true);
    const collapse = tree.find("svg.fa-chevron-down");
    collapse.simulate("click");
    expect(CollapseMock).toHaveBeenCalled();
  });

  it("Doesn't render alertmanager badges when collapsed and only a single cluster is present", () => {
    const tree = MountedSilenceComment(true);
    const ams = tree.find("span.badge.badge-secondary");
    expect(ams).toHaveLength(0);
  });

  it("Doesn't render alertmanager badges when expanded and only a single cluster is present", () => {
    const tree = MountedSilenceComment(false);
    const ams = tree.find("span.badge.badge-secondary");
    expect(ams).toHaveLength(0);
  });

  it("Renders alertmanager badges when collapsed and multiple clusters are present", () => {
    MockMultipleClusters();
    const tree = MountedSilenceComment(true);
    const ams = tree.find("span.badge.badge-secondary");
    expect(ams).toHaveLength(2);
    expect(toDiffableHtml(ams.at(0).html())).toMatch(/default/);
    expect(toDiffableHtml(ams.at(1).html())).toMatch(/fallback/);
  });

  it("Doesn't render alertmanager badges when expanded and multiple clusters are present", () => {
    MockMultipleClusters();
    const tree = MountedSilenceComment(false);
    const ams = tree.find("span.badge.badge-secondary");
    expect(ams).toHaveLength(0);
  });
});
