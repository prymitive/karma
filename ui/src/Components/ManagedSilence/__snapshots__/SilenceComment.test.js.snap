// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SilenceComment /> Matches snapshot when collapsed 1`] = `
"
<div class=\\"d-flex flex-row\\">
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"bell-slash\\"
         class=\\"svg-inline--fa fa-bell-slash fa-w-20 components-managed-silence-icon text-muted\\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 640 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M633.82 458.1l-90.62-70.05c.19-1.38.8-2.66.8-4.06.05-7.55-2.61-15.27-8.61-21.71-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84c-40.33 8.38-74.66 31.07-97.59 62.57L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.35 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.42-6.97 4.17-17.02-2.81-22.45zM157.23 251.54c-8.61 67.96-36.41 93.33-52.62 110.75-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h241.92L157.23 251.54zM320 512c35.32 0 63.97-28.65 63.97-64H256.03c0 35.35 28.65 64 63.97 64z\\"
      >
      </path>
    </svg>
  </div>
  <div class=\\"mx-2 flex-shrink-1 flex-grow-1 mw-1p\\">
    <div class=\\"font-italic components-managed-silence-comment text-truncate overflow-hidden\\">
      Mocked Silence
    </div>
    <div class=\\"components-managed-silence-cite mt-1\\">
      <span class=\\"text-muted mr-2 font-italic\\">
        — me@example.com
      </span>
      <span class=\\"badge badge-light nmb-05 align-text-bottom p-1\\">
        Expires in 30 minutes
        <div class=\\"progress silence-progress\\">
          <div class=\\"progress-bar bg-success\\"
               role=\\"progressbar\\"
               style=\\"width: 50%;\\"
               aria-valuenow=\\"50\\"
               aria-valuemin=\\"0\\"
               aria-valuemax=\\"100\\"
          >
          </div>
        </div>
      </span>
    </div>
  </div>
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <div class=\\"d-flex flex-column flex-sm-row justify-content-between align-items-center\\">
      <div style=\\"display: inline-block; max-width: 100%;\\"
           class=\\" tooltip-trigger\\"
      >
        <span class=\\" badge-primary badge-pill components-label-with-hover components-label badge\\"
              style=\\"opacity: 1;\\"
        >
          123
        </span>
      </div>
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"chevron-down\\"
           class=\\"svg-inline--fa fa-chevron-down fa-w-14 fa-rotate-180 components-managed-silence-icon undefined ml-sm-2 ml-auto mr-sm-0 mr-1 text-muted cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 448 512\\"
           style=\\"transition: transform 0.25s ease-in-out;\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z\\"
        >
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`<SilenceComment /> Matches snapshot when collapsed and multiple clusters are present 1`] = `
"
<div class=\\"d-flex flex-row\\">
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"bell-slash\\"
         class=\\"svg-inline--fa fa-bell-slash fa-w-20 components-managed-silence-icon text-muted\\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 640 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M633.82 458.1l-90.62-70.05c.19-1.38.8-2.66.8-4.06.05-7.55-2.61-15.27-8.61-21.71-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84c-40.33 8.38-74.66 31.07-97.59 62.57L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.35 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.42-6.97 4.17-17.02-2.81-22.45zM157.23 251.54c-8.61 67.96-36.41 93.33-52.62 110.75-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h241.92L157.23 251.54zM320 512c35.32 0 63.97-28.65 63.97-64H256.03c0 35.35 28.65 64 63.97 64z\\"
      >
      </path>
    </svg>
  </div>
  <div class=\\"mx-2 flex-shrink-1 flex-grow-1 mw-1p\\">
    <div class=\\"font-italic components-managed-silence-comment text-truncate overflow-hidden\\">
      Mocked Silence
    </div>
    <div class=\\"components-managed-silence-cite mt-1\\">
      <span class=\\"text-muted mr-2 font-italic\\">
        — me@example.com
      </span>
      <span class=\\"badge badge-secondary mx-1 align-text-bottom p-1\\">
        ha
      </span>
      <span class=\\"badge badge-light nmb-05 align-text-bottom p-1\\">
        Expires in 30 minutes
        <div class=\\"progress silence-progress\\">
          <div class=\\"progress-bar bg-success\\"
               role=\\"progressbar\\"
               style=\\"width: 50%;\\"
               aria-valuenow=\\"50\\"
               aria-valuemin=\\"0\\"
               aria-valuemax=\\"100\\"
          >
          </div>
        </div>
      </span>
    </div>
  </div>
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <div class=\\"d-flex flex-column flex-sm-row justify-content-between align-items-center\\">
      <div style=\\"display: inline-block; max-width: 100%;\\"
           class=\\" tooltip-trigger\\"
      >
        <span class=\\" badge-primary badge-pill components-label-with-hover components-label badge\\"
              style=\\"opacity: 1;\\"
        >
          123
        </span>
      </div>
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"chevron-down\\"
           class=\\"svg-inline--fa fa-chevron-down fa-w-14 fa-rotate-180 components-managed-silence-icon undefined ml-sm-2 ml-auto mr-sm-0 mr-1 text-muted cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 448 512\\"
           style=\\"transition: transform 0.25s ease-in-out;\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z\\"
        >
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`<SilenceComment /> Matches snapshot when collapsed and multiple clusters are present 2`] = `
"
<div class=\\"d-flex flex-row\\">
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"bell-slash\\"
         class=\\"svg-inline--fa fa-bell-slash fa-w-20 components-managed-silence-icon text-muted\\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 640 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M633.82 458.1l-90.62-70.05c.19-1.38.8-2.66.8-4.06.05-7.55-2.61-15.27-8.61-21.71-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84c-40.33 8.38-74.66 31.07-97.59 62.57L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.35 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.42-6.97 4.17-17.02-2.81-22.45zM157.23 251.54c-8.61 67.96-36.41 93.33-52.62 110.75-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h241.92L157.23 251.54zM320 512c35.32 0 63.97-28.65 63.97-64H256.03c0 35.35 28.65 64 63.97 64z\\"
      >
      </path>
    </svg>
  </div>
  <div class=\\"mx-2 flex-shrink-1 flex-grow-1 mw-1p\\">
    <div class=\\"font-italic components-managed-silence-comment \\">
      Mocked Silence
    </div>
    <div class=\\"components-managed-silence-cite mt-1\\">
      <span class=\\"text-muted mr-2 font-italic\\">
        — me@example.com
      </span>
    </div>
  </div>
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <div class=\\"d-flex flex-column flex-sm-row justify-content-between align-items-center\\">
      <div style=\\"display: inline-block; max-width: 100%;\\"
           class=\\" tooltip-trigger\\"
      >
        <span class=\\" badge-primary badge-pill components-label-with-hover components-label badge\\"
              style=\\"opacity: 1;\\"
        >
          123
        </span>
      </div>
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"chevron-down\\"
           class=\\"svg-inline--fa fa-chevron-down fa-w-14 components-managed-silence-icon undefined ml-sm-2 ml-auto mr-sm-0 mr-1 text-muted cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 448 512\\"
           style=\\"transition: transform 0.25s ease-in-out;\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z\\"
        >
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`<SilenceComment /> Matches snapshot when expanded 1`] = `
"
<div class=\\"d-flex flex-row\\">
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"bell-slash\\"
         class=\\"svg-inline--fa fa-bell-slash fa-w-20 components-managed-silence-icon text-muted\\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 640 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M633.82 458.1l-90.62-70.05c.19-1.38.8-2.66.8-4.06.05-7.55-2.61-15.27-8.61-21.71-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84c-40.33 8.38-74.66 31.07-97.59 62.57L45.47 3.37C38.49-2.05 28.43-.8 23.01 6.18L3.37 31.45C-2.05 38.42-.8 48.47 6.18 53.9l588.35 454.73c6.98 5.43 17.03 4.17 22.46-2.81l19.64-25.27c5.42-6.97 4.17-17.02-2.81-22.45zM157.23 251.54c-8.61 67.96-36.41 93.33-52.62 110.75-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h241.92L157.23 251.54zM320 512c35.32 0 63.97-28.65 63.97-64H256.03c0 35.35 28.65 64 63.97 64z\\"
      >
      </path>
    </svg>
  </div>
  <div class=\\"mx-2 flex-shrink-1 flex-grow-1 mw-1p\\">
    <div class=\\"font-italic components-managed-silence-comment \\">
      Mocked Silence
    </div>
    <div class=\\"components-managed-silence-cite mt-1\\">
      <span class=\\"text-muted mr-2 font-italic\\">
        — me@example.com
      </span>
    </div>
  </div>
  <div class=\\"flex-shrink-0 flex-grow-0\\">
    <div class=\\"d-flex flex-column flex-sm-row justify-content-between align-items-center\\">
      <div style=\\"display: inline-block; max-width: 100%;\\"
           class=\\" tooltip-trigger\\"
      >
        <span class=\\" badge-primary badge-pill components-label-with-hover components-label badge\\"
              style=\\"opacity: 1;\\"
        >
          123
        </span>
      </div>
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"chevron-down\\"
           class=\\"svg-inline--fa fa-chevron-down fa-w-14 components-managed-silence-icon undefined ml-sm-2 ml-auto mr-sm-0 mr-1 text-muted cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 448 512\\"
           style=\\"transition: transform 0.25s ease-in-out;\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z\\"
        >
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;
